src/hotspot/cpu/x86/c1_CodeStubs_x86.cpp

```diff
@@ -327,9 +327,11 @@ void PatchingStub::emit_code(LIR_Assembler* ce) {
       assert(a_byte == *start++, "should be the same code");
     }
 #endif
-  } else if (_id == load_mirror_id) {
-    // produce a copy of the load mirror instruction for use by the being
-    // initialized case
+  } else if (_id == load_mirror_id || _id == load_appendix_id) {
+    // Produce a copy of the load mirror / appendix instruction.
+    // 8-byte align the address of the oop immediate to guarantee atomicity
+    // when patching since the GC might walk nmethod oops concurrently.
+    __ align(8, __ offset() + NativeMovConstReg::data_offset_rex);
```

Comment:
In 32-bit VM oops are 4 bytes so 8 bytes is overkill but I am fine with unified alignment.
Should we align mov_metadata() too or it is guarantee aligned already?

Reply:
I don't think we need to guarantee atomicity for metadata because it's not observed concurrently as far as I know, right?